Sources/Exporters/OpenTelemetryProtocol/metric/OltpHTTPMetricExporter.swift

```diff
@@ -0,0 +1,63 @@
+//
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import OpenTelemetrySdk
+import Foundation
+
+public func defaultOltpHTTPMetricsEndpoint() -> URL {
+    URL(string: "http://localhost:4318/v1/metrics")!
+}
+
+public class OtlpHttpMetricExporter: MetricExporter {
+    let endpoint: URL
+    private let httpClient: HTTPClient
+    var pendingMetrics: [Metric] = []
+    
+    public init(endpoint: URL = defaultOltpHTTPMetricsEndpoint(), useSession: URLSession? = nil) {
+        self.endpoint = endpoint
+        if let providedSession = useSession {
+            self.httpClient = HTTPClient(session: providedSession)
+        } else {
+            self.httpClient = HTTPClient()
+        }
+    }
+    
+    public func export(metrics: [Metric], shouldCancel: (() -> Bool)?) -> MetricExporterResultCode {
+        pendingMetrics.append(contentsOf: metrics)
+        return self.flush()
+    }
+
+    public func flush() -> MetricExporterResultCode {
+        var exporterResult: MetricExporterResultCode = .success
+
+        let metrics = pendingMetrics
+        pendingMetrics = []
+        let body = Opentelemetry_Proto_Collector_Metrics_V1_ExportMetricsServiceRequest
+            .with {
+                $0.resourceMetrics = MetricsAdapter.toProtoResourceMetrics(metricDataList: metrics)
+            }
+
+        var request = URLRequest(url: endpoint)
+        request.httpMethod = "POST"
+        request.httpBody = try? body.serializedData()
+        request.setValue("application/x-protobuf", forHTTPHeaderField: "Content-Type")
+        
+        httpClient.send(request: request) { result in
+            switch result {
+            case .success(_):
+                exporterResult = MetricExporterResultCode.success
+            case .failure(let error):
+                print("ERROR: \(error)")
+                exporterResult = MetricExporterResultCode.failureNotRetryable // FIXME how do I know what type of failure?
+            }
+        }
+        return exporterResult
```

Comment:
I'm not sure I get what's happening here. exporterResult is set in the async block, but the value will be returned before the network request has completed?

We could get around that using a semaphore which would stop the return from being called until the network request has finished but that's not what we have currently (it may help us get rid of the 'sleep' in HTTPClient though)

Thoughts Nacho?

Reply:
Yes, the sleep in HTTPClient makes no sense, we should not need to synchronize after the sending by default, only maybe on shutdown